docs/provisioning-methods-and-credentialszip.adoc

```diff
@@ -0,0 +1,82 @@
+// Copy of this topic: https://docs.ota.here.com/concepts/provisioning-methods-and-credentialszip.html adapted for aktualizr repo. Replaces duplicate "credentials.adoc".
+
+= Provisioning methods and credentials.zip
+
+If you're trying to work on integrating {product-name} into your device build, it may be helpful to have a little bit of reference information on how the different provisioning methods work together, and exactly what components are inside `credentials.zip`.
+
+== `credentials.zip` file format
+
+The following files are present in credentials.zip:
+
+// tag::credentials-table[]
+
+[options="header"]
+|======================
+| Filename in zip | Purpose | Used by
+| treehub.json | Location and authentications for treehub and Uptane repo | garage-sign, garage-push
+| client.crt | Certificate for TLS client authentication | garage-push
+| client.key | Private key for TLS client authentication | garage-push
+| root.crt | Root CA for TLS client authentication | garage-push
+| autoprov_credentials.p12 | TLS client credentials that are required when provisioning devices with shared credentials | aktualizr, aktualizr-cert-provider
+| autoprov.url | URL for provisioning server | aktualizr, aktualizr-cert-provider
+| root.json | Initial Uptane root.json (for secure bootstrapping) | garage-sign
+| targets.pub | Public key for offline Uptane image signing | garage-sign
+| targets.sec | Private key for offline Uptane image signing | garage-sign
+| tufrepo.url | URL to Uptane repository | garage-sign
+|======================
+
+// end::credentials-table[]
+
+As you can see, the relevant files for the device itself are `autoprov_credentials.p12` and `autoprov.url`.
+
+When you link:./provision-with-device-credentials.adoc#use-a-hardware-security-module-hsm-when-provisioning-with-device-credentials[Use a Hardware Security Module (HSM)] to provisioning with device credentials, the `implicit_writer` function in meta-updater takes over. The initial credentials on the device won't be valid; this is why you need to copy in the generated ones after booting it.
+
+
+== Configuration options for provisioning with shared credentials
+
+When provisioning with shared credentials, the process works as follows:
+
+1. The user downloads a package that contains an initial provisioning key. This is installed on all devices and is only used to bootstrap the provisioning process.
+2. When Aktualizr first runs on a device, it uses this key/certificate to make a request to the device gateway https API.  The server registers the new device and returns a PKCS#12 archive containing the real credentials for the device.
+3. The device unpacks the root CA certificate and client certificate from this bundle into the paths specified in the `[storage]` section of the config.
```

Comment:
This might not be your fault, but actually they don't get unpacked anywhere except in memory. I would say instead "the device reads the root CA certificate and client certificate from this bundle".

Reply:
@patrickvacek I fixed all the issues that I understood. I fixed the broken links, the file name, and stripped out unnecessary comments.

Sorry for being a dunce, but in some instances when you say stuff is "still broken" and "still valid" I don't know what you're referring to. In many cases the content looks fine to me. Basically all the issues that don't concern broken links or comments.